src/finding/summary.go

```diff
@@ -137,3 +164,56 @@ func findingAttributeValue(finding types.ImageScanFinding, name string) string {
 	}
 	return ""
 }
+
+const legacyCVEURL = "https://cve.mitre.org/cgi-bin/cvename.cgi?name="
+const updatedCVEURL = "https://www.cve.org/CVERecord?id="
+
+func fixFindingURI(name string, uri string) string {
+	correctedURI := uri
+
+	// transition from the old CVE site that is deprecated
+	if strings.HasPrefix(correctedURI, legacyCVEURL) {
+		correctedURI = strings.Replace(correctedURI, legacyCVEURL, updatedCVEURL, 1)
+	}
+
+	// sometimes links are published that are not valid: in this case point to a
+	// GH vuln search as a way to provide some value
+	if strings.HasPrefix(correctedURI, updatedCVEURL) && !strings.HasPrefix(name, "CVE-") {
+		correctedURI = "https://github.com/advisories?query=" + url.QueryEscape(name)
```

Comment:
This is a search, rather than a direct link to the CVE, so it's not as good a user experience. This is why we only do it when the link is invalid.

Also, the direct links don't have the CVE in the URL path: we'd have to look up the issue with GH and create the URL.